be/src/exec/es/es_predicate.cpp

```diff
@@ -384,10 +390,27 @@ Status EsPredicate::build_disjuncts_list(const Expr* conjunct) {
         _disjuncts.push_back(predicate);
 
         return Status::OK();
-    } 
-    
+    }
     if (TExprNodeType::COMPOUND_PRED == conjunct->node_type()) {
         if (TExprOpcode::COMPOUND_OR != conjunct->op()) {
+            // processe COMPOUND_AND, such as:
+            // k = 1 or (k1 = 7 and (k2 in (6,7) or k3 = 12))
+            // k1 = 7 and (k2 in (6,7) or k3 = 12) is compound pred, we should rebuild this sub tree
+            if (TExprOpcode::COMPOUND_AND == conjunct->op()) {
```

Comment:
There are three possibilities here. AND, OR, NOT, For me, `NOT` can be supported too.
And you can do as follow

```
if (conjunct->op() == AND)
else if (conjunct->op() == NOT)

DCHECK(op() == OR)
// handle OR
```